tchannel/transport.py

```diff
@@ -0,0 +1,41 @@
+from __future__ import (
+    absolute_import, division, print_function, unicode_literals
+)
+
+from . import schemes
+
+CALLER_NAME = "cn"
+CLAIM_AT_START = "cas"
+CLAIM_AT_FINISH = "caf"
+FAILURE_DOMAIN = "fd"
+RETRY_FLAGS = "re"
+SCHEME = "as"
+SHARD_KEY = "sk"
+SPECULATIVE_EXE = "se"
+
+
+def to_kwargs(data):
+
+    args = {}
+    args['caller_name'] = data.get(CALLER_NAME)
+    args['claim_at_start'] = data.get(CLAIM_AT_START)
+    args['claim_at_finish'] = data.get(CLAIM_AT_FINISH)
+    args['failure_domain'] = data.get(FAILURE_DOMAIN)
+    args['retry_flags'] = data.get(RETRY_FLAGS)
+    args['scheme'] = data.get(SCHEME)
+    args['shard_key'] = data.get(SHARD_KEY)
+    args['speculative_exe'] = data.get(SPECULATIVE_EXE)
+
```

Comment:
Going to ship it this way - I don't see an immediate need to allow user-defined transport headers, especially since I want the response to end up in a well-defined object